source/language/types.rst

```diff
@@ -180,6 +180,20 @@ types for the purposes of casting.
    // Declare a machine-precision float.
    float my_machine_float = 2.3;
 
+.. _void-type:
+
+Void type
+~~~~~~~~~
+
+Subroutines and externs that do not need to return a value may be declared to
+return ``void``. The ``void`` type is unrealizable and uninstantiable, and thus
+cannot be attached to an identifer.
```

Comment:
Shall we say "or used as a cast operator" as well for completeness?  It's already implied by "uninstantiable", so the addition may be unnecessary.